packages/fxa-shared/metrics/amplitude.ts

```diff
@@ -159,6 +159,48 @@ function mapDomainValidationResult(
   return;
 }
 
+function mapSubscriptionEventProperties(
+  eventType: string,
+  eventCategory: string,
+  eventTarget: string,
+  data: EventData
+) {
+  if (data) {
+    const properties: { [key: string]: string } = {};
+    if (data.country_code_source) {
+      properties['country_code_source'] = data.country_code_source;
+    }
+
+    if (data.payment_provider) {
+      properties['payment_provider'] = data.payment_provider;
+    }
+
+    if (data.plan_id) {
+      properties['plan_id'] = data.plan_id;
+    }
+
+    if (data.product_id) {
+      properties['product_id'] = data.product_id;
+    }
+
+    if (data.provider_event_id) {
+      properties['provider_event_id'] = data.provider_event_id;
+    }
+
+    if (data.subscription_id) {
+      properties['subscription_id'] = data.subscription_id;
+    }
+
+    if (data.voluntary_cancellation) {
+      properties['voluntary_cancellation'] = data.voluntary_cancellation;
+    }
+
+    return properties;
```

Comment:
This could probably be replaced by importing `pick` from lodash:

```typescript
const keys =  ['country_code_source', 'payment_provider', 'plan_id', 'product_id', 'provider_event_id', 'subscription_id', 'voluntary_cancellation'];
return pick(data, keys)
```